dartagnan/src/main/java/com/dat3m/dartagnan/witness/WitnessBuilder.java

```diff
@@ -109,22 +112,22 @@ public WitnessGraph build() {
 			return graph;
 		}
 
-		FormulaManager m = context.getFormulaManager();
 		try (Model model = prover.getModel()) {
 			List<Event> execution = reOrderBasedOnAtomicity(context.getTask().getProgram(), getSCExecutionOrder(model));
 
 			for (int i = 0; i < execution.size(); i++) {
 				Event e = execution.get(i);
 				if (i + 1 < execution.size()) {
 					Event next = execution.get(i + 1);
-					if (e.getCLine() == next.getCLine() && e.getThread() == next.getThread()) {
+					if (e.hasEqualMetadata(next, SourceLocation.class) && e.getThread() == next.getThread()) {
 						continue;
 					}
 				}
 
+				int cLine = e.hasMetadata(SourceLocation.class) ? e.getMetadata(SourceLocation.class).getLineNumber() : -1;
```

Comment:
why not `@unknown` instead of -1?

Reply:
This is the original code in the `WitnessBuilder`. I have not checked what it is used for and if it has to be an int, so I kept it as `-1`.
If we can use arbitrary strings, then I agree that `unknown` is the better option.

Reply:
According to [the witness format](https://github.com/sosy-lab/sv-witnesses/blob/main/README-GraphML.md): 

> The attr.type attribute of this key is int

Thus `-1` is correct